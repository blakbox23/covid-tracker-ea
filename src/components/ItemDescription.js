import React from 'react';

function ItemDescription() {
  return (
    <div>
      <p>I am not known </p>
    </div>
  );
}

export default ItemDescription;
