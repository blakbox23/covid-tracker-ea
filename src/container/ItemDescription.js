/*eslint-disable*/
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getCountryData } from '../redux/countries/countries';
import Nav from '../components/Nav';

function ItemDescription(props) {
  const country = props.country
  const dispatch = useDispatch();
  const countryStats = useSelector((state) => state.data.countryData);
  // console.log("Country stats");
  // console.log(countryStats);

  useEffect(() => {
        dispatch(getCountryData(country));
      }, []);

  return (
    <div>
      <Nav txt={'Details'}/>
    <header className="header flex">
        <div className="header-img">
        {/* <img className="map" src= {`https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/${countryStats.id.toLowerCase()}/vector.svg`} ></img> */}
        {/* {countryStats.id.toLowerCase()} */}
        </div>
        <div className="header-text">
          <div><b>{country}</b></div>

          {(countryStats.totalcases)}         
        </div>
      </header>

      <p className="title">COVID NUMBERS IN {country} </p>

      <div className="stats-list">
      <div className="flex stats-list-item" ><span>Total Cases:</span> <span>{(countryStats.totalcases)}</span></div>
      <div className="flex stats-list-item" ><span>Recovered:</span> <span>{(countryStats.recovered)}</span></div>
      <div className="flex stats-list-item" ><span>Active:</span> <span>{(countryStats.active)}</span></div>
      <div className="flex stats-list-item" ><span>Critical:</span> <span>{(countryStats.critical)}</span></div>
      <div className="flex stats-list-item" ><span>Tests:</span> <span>{(countryStats.tests)}</span></div>
      <div className="flex stats-list-item" ><span>Deaths:</span> <span>{(countryStats.deaths)}</span></div>
      </div>
    
    </div>
  );
}

export default ItemDescription;
