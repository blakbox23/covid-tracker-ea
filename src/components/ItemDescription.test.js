import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import ItemDescription from './ItemDescription';
import store from '../redux/configureStore';
import { BrowserRouter as Router } from 'react-router-dom';


const country = 'Kenya';
it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
        <Provider store={store}>
          <ItemDescription country={country} />
        </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });