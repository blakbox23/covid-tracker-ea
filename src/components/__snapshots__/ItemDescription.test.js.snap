// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="navbar flex"
  >
    <div>
      <a
        className="back-btn"
        href="/"
        onClick={[Function]}
      >
        &lt;
      </a>
    </div>
    <div
      className="logo-txt"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Covid in Africa
      </a>
    </div>
    <a
      className="logo"
      href="/info"
      onClick={[Function]}
    >
      <div>
        <svg
          className="src-logo"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zM11 7h2v2h-2V7zm0 4h2v6h-2v-6z"
            />
          </g>
        </svg>
      </div>
    </a>
  </div>
  <header
    className="header flex"
  >
    <div
      className="header-img"
    />
    <div
      className="header-text"
    >
      <div>
        <b>
          Kenya
        </b>
      </div>
    </div>
  </header>
  <p
    className="title"
  >
    COVID NUMBERS IN 
    Kenya
     
  </p>
  <div
    className="stats-list"
  >
    <div
      className="flex stats-list-item"
    >
      <span>
        Total Cases:
      </span>
       
      <span />
    </div>
    <div
      className="flex stats-list-item"
    >
      <span>
        Recovered:
      </span>
       
      <span />
    </div>
    <div
      className="flex stats-list-item"
    >
      <span>
        Active:
      </span>
       
      <span />
    </div>
    <div
      className="flex stats-list-item"
    >
      <span>
        Critical:
      </span>
       
      <span />
    </div>
    <div
      className="flex stats-list-item"
    >
      <span>
        Tests:
      </span>
       
      <span />
    </div>
    <div
      className="flex stats-list-item"
    >
      <span>
        Deaths:
      </span>
       
      <span />
    </div>
  </div>
</div>
`;
